linode/manager
# feat: [UIE-8133] - IAM RBAC: add new permissions component

## Description 📝

IAM RBAC - new permissions component. This component currently uses mock data for demonstration purposes. The mock data will be removed in the next PR for the `Assign Permission component`.


## Changes  🔄

- new  permissions component

## Target release date 🗓️

1/14/25 (dev)


## Preview 📷

**Include a screenshot or screen recording of the change.**

:lock: Use the [Mask Sensitive Data](https://cloud.linode.com/profile/settings) setting for security.

:bulb: Use `<video src="" />` tag when including recordings in table.

| Before  | After   |
| ------- | ------- |
| 📷 |![image](https://github.com/user-attachments/assets/c4e2eada-5424-4642-a16a-3f32bb057b70)|

## How to test 🧪

### Prerequisites

(How to setup test environment)

- Ensure the Identity and Access Beta flag is enabled in dev tools
- Click on the username in the users table and go to the tab Assigned Roles or click on the user's menu View User Roles


### Verification steps

(How to verify changes)

- [x] Confirm permissions component renders.
- [x] Confirm component expands and collapses by clicking on the button

<details>
<summary> Author Checklists </summary>

## As an Author, to speed up the review process, I considered 🤔

👀 Doing a self review
❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
🤏 Splitting feature into small PRs
➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
🧪 Providing/improving test coverage
 🔐 Removing all sensitive information from the code and PR description
🚩 Using a feature flag to protect the release
👣 Providing comprehensive reproduction steps
📑 Providing or updating our documentation
🕛 Scheduling a pair reviewing session
📱 Providing mobile support
♿  Providing accessibility support

<br/>


## As an Author, before moving this PR from Draft to Open, I confirmed ✅


</details>
